certbot-nginx/certbot_nginx/_internal/http_01.py

```diff
@@ -172,34 +173,35 @@ def _location_directive_for_achall(self, achall):
         return location_directive
 
 
-    def _make_or_mod_server_block(self, achall):
-        """Modifies a server block to respond to a challenge.
+    def _make_or_mod_server_block(self, achall: achallenges.KeyAuthorizationAnnotatedChallenge
+                                  ) -> Optional[list]:
```

Comment:
Other than this one remaining `list`, this LGTM!
```suggestion
                                  ) -> Optional[List]:
```

Reply:
Oops. A pain to grep for :cry:.